src/components/ListView/ListView.js

```diff
@@ -0,0 +1,180 @@
+import React, {
+  Component,
+  Fragment,
+} from 'react';
+import PropTypes from 'prop-types';
+import queryString from 'query-string';
+import {
+  get,
+  noop,
+} from 'lodash';
+
+import { IntlConsumer } from '@folio/stripes/core';
+
+import {
+  checkboxListShape,
+  trimSearchTerm,
+  withCheckboxList,
+} from '../../utils';
+import { SearchAndSort } from '../SearchAndSort';
+import { ViewContainer } from '../ViewContainer';
+import { listTemplate } from '../ListTemplate';
+import { ActionMenu } from '../ActionMenu';
+
+@withCheckboxList
+export class ListView extends Component {
+  static propTypes = {
+    resources: PropTypes.object.isRequired,
+    mutator: PropTypes.object.isRequired,
+    location: PropTypes.shape({ search: PropTypes.string.isRequired }).isRequired,
+    match: PropTypes.shape({ path: PropTypes.string.isRequired }).isRequired,
+    history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
+    label: PropTypes.node.isRequired,
+    selectedEntity: PropTypes.object.isRequired,
+    checkboxList: checkboxListShape,
+    setList: PropTypes.func.isRequired,
+    showSingleResult: PropTypes.bool,
+    EntityView: PropTypes.func.isRequired,
```

Comment:
I am not sure if `EntityView` name is meaningful enough. Could you please rename it to something like `EntityDetailsView`? Or name this props the same how they are named in SearchAndSort component - ViewRecordComponent and EditRecordComponent

Reply:
Done